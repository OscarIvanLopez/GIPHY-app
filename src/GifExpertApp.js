import React, { useState } from "react";

export const GifExpertApp = () => {
  //   const categories = ["One Punsh", "Samurai X", "Dragon Ball"];
  const [categories, setCategories] = useState([
    "One Punsh",
    "Samurai X",
    "Dragon Ball",
  ]);

  const handleAdd = () => {
    
    setCategories(["Asi es",...categories])
  };
  return (
    <div>
      <h2>GifExpertApp</h2>
      <hr />
      <button onClick={handleAdd}>agregar</button>
      <ol>
        {categories.map((cat) => {
          return <li key={cat}>{cat}</li>;
        })}
      </ol>
    </div>
  );
};
