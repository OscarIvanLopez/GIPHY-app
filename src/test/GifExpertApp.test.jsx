import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("pruebas en el componente <GifExpertApp />", () => {
  test("debe de mostrar correctamente el componente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const categories = ["street fighter", "call of duty warzone"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
