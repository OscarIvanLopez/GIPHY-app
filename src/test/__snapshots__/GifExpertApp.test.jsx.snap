// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <GifExpertApp /> debe de mostrar correctamente el componente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="street fighter"
      key="street fighter"
    />
    <GifGrid
      category="call of duty warzone"
      key="call of duty warzone"
    />
  </ol>
</div>
`;
